x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -241,28 +421,91 @@ export class AlertsClient {
     }
   }
 
-  public async getAuthorizedAlertsIndices(featureIds: string[]): Promise<string[] | undefined> {
-    const augmentedRuleTypes = await this.authorization.getAugmentedRuleTypesWithAuthorization(
-      featureIds,
-      [ReadOperations.Find, ReadOperations.Get, WriteOperations.Update],
-      AlertingAuthorizationEntity.Alert
-    );
+  public async bulkUpdate<Params extends AlertTypeParams = never>({
+    ids,
+    query,
+    index,
+    status,
+  }: BulkUpdateOptions<Params>) {
+    if (ids != null) {
+      // blow up
+      return this.fetchAlertAuditOperate({
+        ids,
+        status,
+        indexName: index,
+        operation: WriteOperations.Update,
+      });
+    } else if (query != null) {
+      try {
+        // execute either a query with ids or
+        // query to be executed in updateByQuery
+        // audit results of that query
+        const fetchAndAuditResponse = await this.queryAndAuditAllAlerts({
+          query,
+          index,
+          operation: WriteOperations.Update,
+        });
+
+        if (!fetchAndAuditResponse?.auditedAlerts) {
+          throw Boom.unauthorized('Failed to audit alerts');
+        }
 
-    // As long as the user can read a minimum of one type of rule type produced by the provided feature,
-    // the user should be provided that features' alerts index.
-    // Limiting which alerts that user can read on that index will be done via the findAuthorizationFilter
-    const authorizedFeatures = new Set<string>();
-    for (const ruleType of augmentedRuleTypes.authorizedRuleTypes) {
-      authorizedFeatures.add(ruleType.producer);
+        const result = await this.esClient.updateByQuery({
```

Comment:
It looks like I misunderstood how this bit functions.

In a nutshell, it fetches the rules, checks authorization, adds audit events, builds the query, _then_ runs `updateByQuery`. So I don't think anything here needs to be changed after all! 🎉 

Restating my other comment, it would be great to verify each of these cases with unit tests to make sure that when a user is unauthorized, the audit events get recorded and the update doesn't actually happen :+1: